scripts/alpine-fips.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ash
+
+set -e
+
+if [[ "$OPENSSL_FIPS" == "1" ]]; then
+
+  # This FIPS installation follows the instructions from the official OpenSSL FIPS User Guide
+  # https://openssl-library.org/source/fips-doc/openssl-3.0.9-security-policy-2024-01-12.pdf
+
+  echo "Enabling FIPS"
+
+  # Install required packages
+  apk add --no-cache --virtual .build-deps wget make gcc libgcc musl-dev linux-headers perl vim
+
+  # Temporary build directory
+  mkdir -p /ossl
+  cd /ossl
+
+  # Download and verify the FIPS module
+  wget --quiet https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz
+  echo "$OPENSSL_HASH openssl-$OPENSSL_VERSION.tar.gz" | sha256sum -c - | grep OK
+  tar -xzf openssl-$OPENSSL_VERSION.tar.gz
+
+  # Build the fips modules
+  cd openssl-$OPENSSL_VERSION
+  ./Configure enable-fips --libdir=lib --prefix=/usr
+  make
+
+  # Install fips enabled openssl
+  make install_fips
+
+  # Update to relink binaries
+  apk upgrade --no-cache -U
```

Comment:
is it possible that this apk upgrade could cause us to re-install a non-fips build of openssl?

Reply:
`-U` is an alias for `--update-cache` and using `--no-cache` I believe will force it to do nothing but relink. I cant actually find any info on if this is actually super necessary - Im going to see if we can just remove it.

Reply:
I've tested it without this and found its unneeded, so I've removed it.